src/fontra/views/editor/editor.js

```diff
@@ -783,13 +854,13 @@ export class EditorController {
     };
   }
 
-  setSelectedTool(toolIdentifier) {
+  setSelectedTool(toolIdentifier, wrapper_id = "unkown") {
```

Comment:
Similarly, can you describe why this argument is needed here? This is a "public" method call, the caller should not need to be aware of any wrapping.

Reply:
Only the wrapper is 'selected' not the subtool, therefor `editToolItem.dataset.tool === toolIdentifier` does not work. This is my solution.

Reply:
You need to solve this differently. The method should only take the tool identifier.

For example, it could detect that it doesn't have a dataset.tool property, and check its childen.

Reply:
We may need to rethink a little bit. I think we need to distinguish between "select this tool or tool group" and "select this subtool". Perhaps the "tool group" needs an identifier, too. Let's chat about how to proceed.

This also affects the comand-<N> shortcuts, too: they currently don't visually work correctly for the shape tool, and I don't think the numbering should "dive into" the shape tool group. I think command-3 should choose the currently chosed shape tool, and 4 should select the ruler.

Reply:
This sounds like a more complex topic. I don't know how to solve this currently.

Reply:
I mark this as resolved as we did a coding session together.